### PR TITLE
Add misspell check and fix all typos

### PR DESCRIPTION
**Pull request by songy23**
Updates #155.

Changes are generated with
$ make misspell-correction

See https://github.com/client9/misspell

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,11 @@ ALL_SRC := $(shell find . -name '*.go' \
                                 -not -path './testbed/*' \
                                 -type f | sort)
 
+# All source code and documents. Used in spell check.
+ALL_SRC_AND_DOC := $(shell find . \( -name "*.md" -o -name "*.go" \) \
```
Comment (pjanotti) [inline Makefile]: Can it do yaml files too? We have comments on those.

File: internal/tools.go [hunk 2]
```diff
@@ -27,4 +27,5 @@ import (
 	_ "github.com/jstemmer/go-junit-report"
 	_ "golang.org/x/lint/golint"
 	_ "golang.org/x/tools/cmd/goimports"
+	_ "github.com/client9/misspell/cmd/misspell"
```
Comment (pjanotti) [inline internal/tools.go]: nit: didn't goimports complain about the order here? :)

File: Makefile (hunk 1 above)
Comment (songy23) [reply Makefile]: Yes, added yaml files.

File: internal/tools.go (hunk 2 above)
Comment (songy23) [reply internal/tools.go]: It did :) fixed in https://github.com/open-telemetry/opentelemetry-service/pull/214/commits/99eabe206252d78b0006b0380930e17fcc3e652a.

Comment (pjanotti) [reply internal/tools.go]: :)

**Pull request merged**